### ISSUE TITLE
REAL runtime 60 hours when running on merged detection files without filtering by SNR or agreement number

### ISSUE THREAD

**Issue by zhengyang-c** [zhengyang-c/cy1400-eqt#14]
Want to run timing tests, on say one month of customfiltered data, varying the grid search size.

Other parameters? Suspect:
- rsel: tolerance for larger residuals of origin times, default is 4
- std0: standard deviation threshold, since REAL finds both the origin time and location at the same time

want to have a program to automatically test REAL for different parameters and quantify:
- run time
- no. of events
- logged outputs (how many suspicious events discarded etc))

This also opens the opportunity to write a more sensible wrapper around REAL, even though it will have a lot of input arguments and honestly that's not that much better
What could help is to have more verbose arguments / write my own documentation after having read the REAL source code?

Honestly C is not easy to read
There will eventually be a need to read the REAL source code to figure out what goes on inside with how it removes `suspicious' events.

Also want to throw in the updated velocity structure, but that's more relevant for hypoDD which I'm not running yet

**Comment by zhengyang-c** [comment]
tl;dr feature request for automated test bench for REAL run time (not sure if over kill but if we'll be using it a lot then it makes sense)

**Comment by zhengyang-c** [comment]
- [ ] Runtime scaling as a function of cells (since this will dominate)
- [ ] No. of events as a function of criteria (residual, residual tolerance, arrival time uncertainties, no. of picks)
- [ ] Shift in locations (per event) as a function of velocity model (use default, or DIY)

**Comment by zhengyang-c** [comment]
https://curc.readthedocs.io/en/latest/programming/OpenMP-C.html

seems to be a need to set the ```OMP_NUM_THREADS``` env var in order for multi-processing to work as stated in the paper

**Comment by zhengyang-c** [comment]
yknow in the paper REAL does VELEST before hypoDD which by the way makes complete sense

**Comment by zhengyang-c** [comment]
3m35s for 32 threads
4m9s for 4 threads
5m34s for 1 thread
R = "2.0/60/0.1/5/3/300/2";

$R = "2.0/60/0.1/5/3/300/2";

**Comment by zhengyang-c** [comment]
Should just email the original author to ask about GCarc0 and nxd

**Comment by zhengyang-c** [comment]
real    712m50.723s
jan 2020, ~40k+ phases input
4 OMP threads

total no. of events?
